docs: Add Depot to GitHub integration page

vikram-dagger opened the pull request.

kylegalbraith commented on docs/current_docs/integrations/depot.mdx:
```diff
@@ -0,0 +1,65 @@
+
+## How it works
+
+When running a CI pipeline on a Dagger-powered Depot GitHub Runner, the general workflow looks like this:
```
```suggestion
When running a CI pipeline on a Dagger-powered Depot GitHub Actions Runner, the general workflow looks like this:
```

kylegalbraith commented on docs/current_docs/integrations/depot.mdx:
```diff
@@ -0,0 +1,65 @@
+When running a CI pipeline on a Dagger-powered Depot GitHub Runner, the general workflow looks like this:
+
+1. GitHub emits a repository event which triggers a workflow
+1. Depot receives this event, process it, and runs one or more jobs in the workflow, each running in an isolated Depot GitHub Runner.
```
```suggestion
1. Depot receives this event, processes it, and runs one or more jobs in the workflow, each running in an isolated Depot GitHub Actions Runner.
```

kylegalbraith commented on docs/current_docs/integrations/depot.mdx:
```diff
@@ -0,0 +1,65 @@
+
+1. GitHub emits a repository event which triggers a workflow
+1. Depot receives this event, process it, and runs one or more jobs in the workflow, each running in an isolated Depot GitHub Runner.
+1. Each Depot GitHub Runner is already connected to an already running Dagger Engine (this exists outside of the Depot GitHub Runner). Previous runs may have cached data in this external Dagger Engine.
```
```suggestion
1. Each Depot GitHub Actions Runner is automatically connected to an already running Dagger Engine (this exists outside of the Depot GitHub Runner). The external Dagger is pre-configured to cache your previous runs out of the box.
```

kylegalbraith commented on docs/current_docs/integrations/depot.mdx:
```diff
@@ -0,0 +1,65 @@
+1. GitHub emits a repository event which triggers a workflow
+1. Depot receives this event, process it, and runs one or more jobs in the workflow, each running in an isolated Depot GitHub Runner.
+1. Each Depot GitHub Runner is already connected to an already running Dagger Engine (this exists outside of the Depot GitHub Runner). Previous runs may have cached data in this external Dagger Engine.
+1. The "Dagger for GitHub" GitHub Action installs the Dagger CLI, and then calls the specified sub-command, module, function name, and arguments.
```
The Dagger CLI is pre-installed into our runner.

vikram-dagger replied:
Fixed

kylegalbraith commented on docs/current_docs/integrations/depot.mdx:
```diff
@@ -0,0 +1,65 @@
+1. Each Depot GitHub Runner is already connected to an already running Dagger Engine (this exists outside of the Depot GitHub Runner). Previous runs may have cached data in this external Dagger Engine.
+1. The "Dagger for GitHub" GitHub Action installs the Dagger CLI, and then calls the specified sub-command, module, function name, and arguments.
+1. The Dagger CLI running inside a Depot GitHub Runner gets automatically connected to Dagger Cloud.
+1. The Dagger CLI is also configured to connect to the external Dagger Engine which is already
```
```suggestion
1. The Dagger CLI is also pre-configured to connect to the external Dagger Engine with persistent caching
```

kylegalbraith commented on docs/current_docs/integrations/snippets/depot-hello.yml:
```diff
@@ -0,0 +1,22 @@
+          # assumes the Dagger Cloud token is in
+          # a repository secret named DAGGER_CLOUD_TOKEN
+          # set via the GitHub UI/CLI
+          cloud-token: ${{ secrets.DAGGER_CLOUD_TOKEN }}
```
I don't believe this is needed with Depot as the cloud token will already be configured via the connection.

vikram-dagger replied:
Fixed

vikram-dagger commented on docs/current_docs/integrations/depot.mdx:
```diff
@@ -0,0 +1,65 @@
+When running a CI pipeline on a Dagger-powered Depot GitHub Actions Runner, the general workflow looks like this:
+
+1. GitHub emits a repository event which triggers a workflow
+1. Depot receives this event, processes it, and runs one or more jobs in the workflow, each running in an isolated Depot GitHub Actions Runner.
```
```suggestion
1. Depot receives this event, processes it, and runs one or more jobs in the workflow, each running in an isolated Depot GitHub Actions Runner. The Depot GitHub Actions Runner has the Dagger CLI pre-installed.
```

kylegalbraith commented on docs/current_docs/integrations/snippets/depot-test-build.yml:
```diff
@@ -0,0 +1,43 @@
+jobs:
+  test:
+    name: test
+    runs-on: depot-ubuntu-latest-*,dagger=0.13.5
```
`-*` is meant to signal your can choose your number of CPUs? I didn't see this called out anywhere in the docs.

vikram-dagger replied:
I wasn't sure what the `*` referred to, I used the syntax shown on https://depot.dev/docs/github-actions/runner-types. I've changed it to be more specific now.

vikram-dagger commented on docs/current_docs/integrations/depot.mdx:
```diff
@@ -0,0 +1,65 @@
+
+1. GitHub emits a repository event which triggers a workflow
+1. Depot receives this event, processes it, and runs one or more jobs in the workflow, each running in an isolated Depot GitHub Actions Runner.
+1. Each Depot GitHub Actions Runner is automatically connected to an already running Dagger Engine (this exists outside of the Depot GitHub Runner). The external Dagger is pre-configured to cache your previous runs out of the box.
```
```suggestion
1. Each Depot GitHub Actions Runner is automatically connected to an already running Dagger Engine (this exists outside of the Depot GitHub Runner). The external Dagger Engine comes with caching pre-configured.
```

shykes commented:
IMO this should be part of a revamped "Github Actions" page. Once we revamp that page and highlight it more (which is something we need to do no matter what, and soon), it will get a lot of traffic. Relegating "Depot" to a separate page, makes it harder to discover Depot unless you already know about it and are explicitly looking for it. Sure we could link to the Depot page from the revamped GHA page, but that feels less seamless and will lead to duplication.

TLDR: the best version of this page IMO is a revamped Github Actions page, which just happens to prominently feature Depot prominently.

vikram-dagger commented:
> TLDR: the best version of this page IMO is a revamped Github Actions page, which just happens to prominently feature Depot prominently.

Thanks for the feedback! I've updated the GitHub page as suggested with tabs for each runner type and removed the standalone Depot page. 

One comment/question: currently the page is titled "GitHub" and not "GitHub Actions", because it also includes other (non-Actions) integration examples, such as pushing to GHCR and using PRs as function arguments. I'm assuming that we will continue with this, because it wouldn't make sense to have two separate integrations pages ie. "GitHub" and "GitHub Actions", but please let me know if you think otherwise.

gerhard commented on docs/current_docs/integrations/snippets/depot-hello.yml:
```diff
@@ -0,0 +1,18 @@
+          version: "0.13.7"
+          verb: call
+          module: github.com/shykes/daggerverse/hello@v0.1.2
+          args: hello --greeting="bonjour" --name="monde"
```
We are missing the following property from all `dagger/dagger-for-github` examples:

```yaml
engine-stop: false
```

We definitely **do not** want to stop this external Engine.

vikram-dagger replied:
Added

shykes commented:
> One comment/question: currently the page is titled "GitHub" and not "GitHub Actions", because it also includes other (non-Actions) integration examples, such as pushing to GHCR and using PRs as function arguments. I'm assuming that we will continue with this, because it wouldn't make sense to have two separate integrations pages ie. "GitHub" and "GitHub Actions", but please let me know if you think otherwise.

I think "Integrating Dagger in your CI" is an important enough topic that it deserves its own section in the docs. And from there, "Github Actions" would be a natural page under that section.

In that context, yes I do think that "Integrating Dagger in Github Actions" deserves its own page separate from other Github integrations. I would design that page with the "Integrating with CI" top-level section in mind. We don't need that top-level section right now, but we need to be able to drop it in later, without breaking the GHA (and therefore Depot) flow.

vikram-dagger commented on docs/current_docs/integrations/snippets/depot-hello.yml:
```diff
@@ -0,0 +1,18 @@
+    runs-on: depot-ubuntu-22.04,dagger=0.13.7
+    steps:
+      -
+        name: Call Dagger Function
```
Fixed

vikram-dagger commented on docs/current_docs/integrations/snippets/depot-hello.yml:
```diff
@@ -0,0 +1,18 @@
+    steps:
+      -
+        name: Call Dagger Function
+        uses: dagger/dagger-for-github@v6
+        with:
+          version: "0.13.7"
```
I'm assuming no change needed here as you mentioned it's not related to this PR - please lmk if I misunderstood

gerhard replied:
That is correct, no change needed here.

jpadams replied:
@gerhard does this capture it?
https://github.com/dagger/dagger-for-github/issues/164

vikram-dagger commented on docs/current_docs/integrations/snippets/depot-test-build.yml:
```diff
@@ -0,0 +1,35 @@
+          version: "0.13.7"
+          verb: call
+          module: github.com/kpenfound/dagger-modules/golang@v0.2.0
+          args: build-container --source=. --args=. publish --address=ttl.sh/my-app-$RANDOM
```
Noted

gerhard commented:
Taking another look now.

gerhard reviewed (approved):
LGTM 👍

samalba commented:
What's the rationale to keep two distincts "Github" and "Github Actions" sections? They feel redundant.

vikram-dagger commented:
> What's the rationale to keep two distincts "Github" and "Github Actions" sections? They feel redundant.

It's discussed above in https://github.com/dagger/dagger/pull/8768#issuecomment-2455340031

samalba commented:
> It's discussed above in [#8768 (comment)](https://github.com/dagger/dagger/pull/8768#issuecomment-2455340031)

Sure. It still feels confusing, but we can always change it later. LGTM overall 👍

shykes commented:
I guess the confusion is from keeping them both in the same generic "Integrations" section, right? @vikram-dagger do you want to move CI integrations into their own section now? We can worry about adding more content to that section later. wdyt @samalba will that help with the confusion? I think @vikram-dagger initially planned to do that later to save time here.

vikram-dagger commented on docs/current_docs/integrations/snippets/depot-hello.yml:
```diff
@@ -0,0 +1,12 @@
+  hello:
+    name: hello
+    runs-on: depot-ubuntu-22.04,dagger=0.13.7
+    steps:
```
@gerhard can you please review this once more?

vikram-dagger commented on docs/current_docs/integrations/snippets/depot-test-build.yml:
```diff
@@ -0,0 +1,23 @@
+  push:
+    branches: [main]
+
+jobs:
```
@gerhard can you please review this once more?

jpadams replied:
@vikram noting that we'll need to add a checkout step to the hello example as well to get it to resolve properly in Dagger Cloud.

vikram-dagger commented:
> I guess the confusion is from keeping them both in the same generic "Integrations" section, right? @vikram-dagger do you want to move CI integrations into their own section now? We can worry about adding more content to that section later. wdyt @samalba will that help with the confusion? I think @vikram-dagger initially planned to do that later to save time here.

I've created https://github.com/dagger/dagger/pull/8866 for the new section. If that helps, then we can merge it and I will rebase this PR.

jpadams commented:
The `hello` example does not make the Dagger Cloud org page switch over to traces view, but leaves it waiting for traces. I was able to go directly to the trace via the link in GHA, but it seems that my trace was an "orphaned trace" which would be a bad first run experience for someone.

![image](https://github.com/user-attachments/assets/376e546f-c99e-41ec-873c-90c65d85c3f1)

I was able to see the `greetings-api` example build live.

Not sure if this is because in the `hello` example, my repo is empty except for the GHA yaml?

cc @vito @crjm

vikram-dagger commented:
> @jpadams @vikram-dagger, here are the links for the videos if you want to embed them in the docs. They are unlisted for now.
> 
> How to set up - https://youtu.be/Y59gmLLOOT8 Demo - https://youtu.be/55fQqPnJ4XU

I've added these to the resources section of the docs. Thank you!

kylegalbraith commented on docs/current_docs/integrations/github-actions.mdx:
```diff
@@ -18,12 +28,47 @@ When running a CI pipeline with Dagger using a standard GitHub Actions runner, t
+
+When running a CI pipeline on a Dagger Powered Depot GitHub Actions Runner, the general workflow looks like this:
+
+1. GitHub emits a repository event which triggers a workflow
```
```suggestion
1. GitHub emits a webhook event which triggers a job in a GitHub Actions workflow
```

kylegalbraith commented on docs/current_docs/integrations/github-actions.mdx:
```diff
@@ -18,12 +28,47 @@ When running a CI pipeline with Dagger using a standard GitHub Actions runner, t
+When running a CI pipeline on a Dagger Powered Depot GitHub Actions Runner, the general workflow looks like this:
+
+1. GitHub emits a repository event which triggers a workflow
+1. Depot receives this event, processes it, and runs one or more jobs in the workflow, each running in an isolated Depot GitHub Actions Runner.
```
```suggestion
1. Depot receives this event, processes it, and launches an isolated Depot GitHub Actions Runner.
```

GitHub sends webhook events for each job in a workflow. A workflow can have multiple jobs, but a runner is only ever allowed to work on one job.

kylegalbraith commented on docs/current_docs/integrations/snippets/depot-hello.yml:
```diff
@@ -0,0 +1,12 @@
+jobs:
+  hello:
+    name: hello
+    runs-on: depot-ubuntu-22.04,dagger={{ version }}
```
```suggestion
    runs-on: depot-ubuntu-24.04,dagger={{ version }}
```
Either one is fine with me. The latest Ubuntu version is faster (regardless of Depot).

jpadams reviewed (approved):
ship it!
